### PR TITLE
Fixes #19: Fix branch protection workflow

### LINKED ISSUE #19: Fix branch protection workflow
The script [branch-protection-ruleset.sh](https://github.com/ai-cfia/devops/blob/main/github-management-script/branch-protection-ruleset.sh) currently adds "Python package / build" which is not a test we have it should be "lint-test / build"

### PR DESCRIPTION
**Pull request by vivalareda**
#19

### CODE REVIEW

File: github-management-script/branch-protection-ruleset.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ set_branch_protection() {
     DATA='{
         "required_status_checks": {
             "strict": true,
-            "contexts": ["Python package / build"]
+            "contexts": ["lint-test / build"]
```
Comment (rngadam) [inline github-management-script/branch-protection-ruleset.sh]: ah, yes, that's the fix I expected

**Review by rngadam — APPROVED** [review]

File: github-management-script/branch-protection-ruleset.sh (hunk 1 above)
Comment (vivalareda) [reply github-management-script/branch-protection-ruleset.sh]: I think it's wrong again looking at the PR from finesse-frontend

Comment (vivalareda) [reply github-management-script/branch-protection-ruleset.sh]: I changed contexts to checks to address this comment
https://github.com/ai-cfia/devops/issues/19#issuecomment-1812649958

Also changed "lint-test / build" to "lint-test / lint-test" reasoning explained there:
https://github.com/ai-cfia/github-workflows/pull/28

**Review by rngadam — APPROVED** [review]

File: github-management-script/branch-protection-ruleset.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ set_branch_protection() {
     DATA='{
         "required_status_checks": {
             "strict": true,
-            "contexts": ["Python package / build"]
+            "checks": ["lint-test / lint-test"]
```
Comment (rngadam) [inline github-management-script/branch-protection-ruleset.sh:25]: we have to discuss what would make these check names "contexts" more meaningful

**Review by rngadam — APPROVED** [review]

**Review by ThomasCardin — APPROVED** [review]
LGTM!

File: github-management-script/branch-protection-ruleset.sh (hunk 2 above)
Comment (rngadam) [reply github-management-script/branch-protection-ruleset.sh:25]: can follow up in a separate PR

**Pull request merged**